### ISSUE TITLE
Use of the TSNet engine with other hydraulic modelling software

### ISSUE THREAD

**Issue by lbutler** [glorialulu/wmoc#10]
Great work on this project!

I'm curious to know if TSNet could be used with modelling package other then EPANET?

The first obvious option would be to convert the model into EPANET.

Alternatively, I was wondering if it would be possible to pass in the definition of the network topology and steady-state results into the transient engine without having to run an EPANET simulation in WNTR to get the initial state?

My limited look through the code base suggests this may be possible, and that you could use an interface/abstraction between WNTR and TSNET which you could inject any modelling results, but I haven't run through the entire project to see where else WNTR is required.

I have models in InfoWater, WaterGEMS and InfoWorks WS Pro that I would like to try this approach with. I also have access to InfoSurge and would also love to compare the results between the two packages and see if we could start using an open-source option in preference to a commercial product.

My questions boil down to:

- Would it be possible to load in other models and their results directly into TSNet without EPANET?
- Is this something you would want to be added into this project?

Thanks again for making this project!

**Comment by linasela** [comment]
Hi Luke,
thanks for trying out TSNet. To your questions:
1. currently, TSNet depends on WNTR for steady state results, network topology, as well as some additional functionalities. Hence, importing models from other software, e.g., WaterGems, InfoWater, Kypipe is not possible. As you mentioned in your comment, these software allow you to export their models as .inp file, and this is what TSNet uses. So at this point, to use TSNet you would need to convert these models to EPANET. At this moment, we are not planning to add this integration.

2. we have done some simulations comparing TSNet with Hammer, not InfoSurge. But to do this, we exported the results manually from Hammer and imported them into python to make the comparison. please note, that commercial software offer more modeling capabilities than current TSNet version. We have a long term vision for TSNet, in which we hope to continue the open source development and adding more functionalities. We are welcoming any contributions, test cases, comments that can help us improve TSNet.

Best,
Lina

**Comment by lbutler** [comment]
Thanks @linasela 

While some models can be converted to EPANET, there are limitations in its hydraulic engine which means a direct conversion is not always simple, especially with EPS models with complex controls. 

I might still try what I've suggested on my own fork of this project. I've got a few modelling projects going with transient loggers in the field and being able to use TSNet directly from the existing model may be beneficial.

If I have any luck, I'll report back, but appreciate it is not a goal of this project.

Best of luck and keep up the good work, I will keep an eye on this project and see how it develops.
